code/modules/mining/equipment/goliath_hide.dm

```diff
@@ -20,13 +20,8 @@
 	if(!proximity_flag)
 		return
 	if(is_type_in_typecache(target, goliath_platable_armor_typecache))
-		var/obj/item/clothing/C = target
-		if(C.armor.melee < 60)
-			C.armor = C.armor.setRating(melee = min(C.armor.melee + 10, 60))
-			to_chat(user, "<span class='info'>You strengthen [target], improving its resistance against melee attacks.</span>")
-			use(1)
-		else
-			to_chat(user, "<span class='warning'>You can't improve [C] any further!</span>")
+		target.LoadComponent(/datum/component/goliath_plate,src,user)
```

Comment:
I looked at it lol.

Needs some sort of check here to determine if `target` already has the component (I want to say there's a helper for it) so you're not constantly getting the component to load.

Reply:
That's what LoadComponent _is_. It GetComponent()s and if that returns null AddComponent()s.

Reply:
Sweet!